Documentation/project-docs/developer-guide.md

```diff
@@ -16,4 +11,235 @@ The CoreFX repo can be built from a regular, non-admin command prompt. The build
+
+The native component should be buildable on any system.
+
+**Examples**
+
+- Building in debug mode for platform x64
+```
+build-native -debug -buildArch=x64
+```
+
+- The following example shows the argument `--`. Everything that is after it is not going to be processed and it is going to be passed as it is.
+```
+build-native -debug -buildArch=arm -- cross verbose
+```
+
+For more information about extra parameters take a look at the scripts `build-native` under src/Native.
+
+##Build Managed
+Since the managed build uses the .NET Core CLI (which the build will download), managed components can only be built on a subset of distros.
+There are some additional pre-requesties from the CLI which need to be installed. Both libicu and
+libunwind are used by CoreCLR to execute managed code, so they must be
+installed. Since CoreFX does not actually link against these packages, runtime
+versions are sufficent.  We also require curl to be present, which we use to
+download the .NET Core CLI.
+
+**Examples**
+
+- Building in debug mode for platform x64
+```
+build-managed -debug -buildArch=x64
+```
+
+- Building in debug mode for platform x64 targeting OS Linux
+```
+build-managed -debug -buildArch=x64 -os=Linux
+```
+
+- Building only the managed libraries with /flp:v=diag. It doesn't restore packages, builds tests or builds packages.
+```
+build-managed -binaries -verbose
+```
+
+- Building the managed libraries and the tests but don't run the tests.
+```
+build-managed -skiptests
+```
+
+- The following example shows the argument `--`. Everything that is after it is not going to be processed and it is going to be passed as it is.
+Use it to pass extra msbuild properties.
+```
+build-managed -binaries -- /p:WithoutCategories=IgnoreForCI
+```
+
+### Building individual CoreFx DLLs
+
+Under the src directory is a set of directories, each of which represents a particular assembly in CoreFX.
+  
+For example the src\System.Diagnostics.DiagnosticSource directory holds the source code for the System.Diagnostics.DiagnosticSource.dll assembly. Each of these directories includes two projects, one for the DLL being built and one for the tests, both specified by a .builds file.
+
+You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the src\System.Diagnostics.DiagnosticsSource\src directory and typing `msbuild System.Diagnostics.DiagnosticSource.builds`. The DLL ends up as  bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource\System.DiagnosticSource.dll.
+
+You can build the tests for System.Diagnostics.DiagnosticSource.dll by going to 
+src\System.Diagnostics.DiagnosticSource\tests and typing `msbuild System.Diagnostics.DiagnosticSource.Tests.builds`.
+
+There is also a pkg directory for each project, and if you go into it and type `msbuild`, it will build the DLL (if needed)
+and then also build the NuGet package for it. The NuGet package ends up in the bin\packages directory.  
```

Comment:
We really should also add some instructions about how to build an individual csproj for the different targets and OS's without needing to rebuild all configurations via the builds file.